Swift/KVSiOSApp/SignalingClient.swift

```diff
@@ -136,15 +136,21 @@ extension SignalingClient: WebSocketDelegate {
                         debugPrint("SDP answer received from signaling \(sdp)")
                     } else if messageType == "ICE_CANDIDATE" {
                         guard let iceCandidate = jsonObject["candidate"] as? String else {
+                            print("Dropping \(jsonObject), due to missing 'candidate' property")
                             return
                         }
-                        guard let sdpMid = jsonObject["sdpMid"] as? String else {
+                        let sdpMid = jsonObject["sdpMid"] as? String
+                        let sdpMLineIndex = jsonObject["sdpMLineIndex"] as? Int32
+                        
+                        guard sdpMid != nil || sdpMLineIndex != nil else {
```

Comment:
is there  a check already for sdpMLineIndex <0 ?

Reply:
google webrtc accepts -1. I checked by hard-coding it. No crashes or exceptions, just won't ever use that ice candidate since the index doesn't match.

It should be responsibility of the webrtc impl. to not generate an ICE candidate with index -1. The purpose of this class is to construct ice candidate object from string representation, which it does successfully. User listens to this callback (didReceiveIceCandidate) and decide if they want to add the `-1` index to the peer connection.